woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -1631,3 +1632,164 @@ def test_sets_koalas_option_on_init(sample_df_koalas):
+
+    if ks and isinstance(sample_df, ks.DataFrame):
+        dtype = 'int64'
+        new_series = pd.Series([1, 2, 3], dtype=dtype)
+        new_series = ks.Series(new_series)
+    else:
+        dtype = 'Int64'
+        new_series = pd.Series([1, 2, 3], dtype=dtype)
+
+    df.ww['test_col2'] = new_series
+    assert 'test_col2' in df.columns
+    assert 'test_col2' in df.ww.columns
+    assert df.ww['test_col2'].ww.logical_type == Integer
+    assert df.ww['test_col2'].ww.semantic_tags == set()
+    assert df.ww['test_col2'].name == 'test_col2'
+    assert df.ww['test_col2'].dtype == dtype
+
+    # Standard tags and no logical type
+    if ks and isinstance(sample_df, ks.DataFrame):
+        dtype = 'object'
+        new_series = pd.Series(['new', 'column', 'inserted'], name='test_col', dtype=dtype)
+        new_series = ks.Series(new_series)
+    else:
+        dtype = 'category'
+        new_series = pd.Series(['new', 'column', 'inserted'], name='test_col', dtype=dtype)
+
+    new_series.ww.init()
+    df.ww['test_col'] = new_series
+    assert 'test_col' in df.columns
+    assert 'test_col' in df.ww.columns
+    assert df.ww['test_col'].ww.logical_type == Categorical
+    assert df.ww['test_col'].ww.semantic_tags == {'category'}
+    assert df.ww['test_col'].name == 'test_col'
+    assert df.ww['test_col'].dtype == dtype
+
+    new_series = pd.Series([1, 2, 3], dtype='float')
+    if ks and isinstance(sample_df, ks.DataFrame):
+        new_series = ks.Series(new_series)
+
+    new_series.ww.init(
+        logical_type=Double,
+        use_standard_tags=False,
+        semantic_tags={'test_tag'},
+    )
+
+    df.ww['test_col3'] = new_series
+    assert 'test_col3' in df.columns
+    assert 'test_col3' in df.ww.columns
+    assert df.ww['test_col3'].ww.logical_type == Double
+    assert df.ww['test_col3'].ww.semantic_tags == {'test_tag'}
+    assert df.ww['test_col3'].name == 'test_col3'
+    assert df.ww['test_col3'].dtype == 'float'
+
+
+def test_setitem_overwrite_column(sample_df):
+    df = sample_df.copy()
+    df.ww.init(
+        index='id',
+        time_index='signup_date',
+        use_standard_tags=True
+    )
+
+    # Change to column no change in types
+    original_col = df['age']
+    original_col.ww.init()
+    new_series = pd.Series([1, 2, 3])
+    if ks and isinstance(sample_df, ks.DataFrame):
+        dtype = 'int64'
+        new_series = ks.Series(new_series)
+    else:
+        dtype = 'Int64'
+        new_series = new_series.astype(dtype)
+
+    new_series.ww.init(use_standard_tags=True)
+    df.ww['age'] = new_series
+
+    assert 'age' in df.columns
+    assert 'age' in df.ww.columns
+    assert df.ww['age'].ww.logical_type == original_col.ww.logical_type
+    assert df.ww['age'].ww.semantic_tags == original_col.ww.semantic_tags
+    assert df.ww['age'].dtype == dtype
+    assert original_col is not df.ww['age']
+
+    # Change dtype, logical types, and tags with conflicting use_standard_tags
```

Comment:
I guess I'm not clear on what we are testing with the conflicting use_standard_tags. I think we would get the same result here if there was no conflict, since the LogicalType Boolean doesn't have any standard tags.

Reply:
Is there anything missing in this replicate of the test? How do we want to update?

Reply:
I think we should use a different LogicalType here. If we use a Double for the new column, we can do the following:
- Create the dataframe with use_standard_tags = True
- Create the series with use_standard_tags = False and add another custom tag like `test_tag` which is currently being done
- Assign the new column to the dataframe and make sure the semantic tags of the column in the dataframe contain the `numeric` tag as well as the custom tag `test_tag`

Reply:
I agree that we want the table's `use_standard_tags` to take precedence here, but I think it'd be worth warning users if we've changed a column's tags in order to add it to a Woodwork table.

And the case nate outlined is one where Woodwork adds tags to a column, but we should also test a case where Woodwork removes tags from a column before adding it 

- Create the dataframe with use_standard_tags = False
- Create the series with use_standard_tags = True and add another custom tag like test_tag which is currently being done
- Assign the new column to the dataframe and make sure the semantic tags of the column in the dataframe contains only the custom tag test_tag